### DISCUSSION TITLE
Combining image and data response in fast api.

### DISCUSSION THREAD

**Discussion by FatemaD1577** [fastapi/fastapi#10463] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

Hi,

I want to send image as well as information related to the image as response in fast api. I tried passing data as header information in file response, I am able to get only image in the output response not the data information. Can anyone help me.

**Comment by hasansezertasan** [comment]
I believe you can turn the image into bytecode and put it inside a dictionary with the related information.

**Reply by FatemaD1577** [reply]
Can you explain with the help of an example code?

**Reply by jgould22** [reply]
> I tried passing data as header information in file response, I am able to get only image in the output response not the data information. 

Can you expand on this a bit. How are you trying to access the headers ? 

A second option is to base64 encode the binary data of the image and place it in a json object.

A third option is to not return the image but instead return the metadata with a link for the client to get the image after they receive the response.

**Reply by hasansezertasan** [reply]
I would go with the third option, I think its easier to maintain.

**Comment by hasansezertasan** [comment, marked as answer]
Here is an example:

```python
import os
import base64

from fastapi import FastAPI

basedir = os.path.abspath(os.path.dirname(__file__))
app = FastAPI()


@app.get("/")
async def index():
    filename = os.path.join(basedir, "example.jpg")
    with open(filename, "rb") as f:
        data = f.read()
    return {
        "name": "example.jpg",
        "type": "image/jpeg",
        "size": len(data),
        "data": base64.b64encode(data),
    }

```

A usefull [stackoverflow question](https://stackoverflow.com/questions/59760739/how-do-i-return-a-dict-an-image-from-a-fastapi-endpoint).

**Comment by glass-ships** [comment]
Can confirm hasan's example works, this is the method I've come across and used recently.
it's worth noting however that this can be quite slow to perform if you're sending multiple files, especially depending on their size, whether its due to having to `.read()` the file or the `b64encode()` is not immediately clear to me. I haven't done much testing yet, but have run into extremely.  slow loading react app which gets about 20 images from FastAPI, with FastAPI being the bottleneck
